third_party/toolchains/rbe_win2019/Dockerfile

```diff
@@ -1,94 +1,79 @@
-# This dockerfile is taken from go/rbe-windows-user-guide
-# (including the fix --compilation_mode=dbg)
-
-# This Dockerfile creates an image that:
-# - Has the correct MTU setting for networking from inside the container to work.
-# - Has Visual Studio 2015 Build Tools installed.
-# - Has msys2 + git, curl, zip, unzip installed.
-# - Has Python 2.7 installed.
-# TODO(jsharpe): Consider replacing "ADD $URI $DEST" with "Invoke-WebRequest -Method Get -Uri $URI -OutFile $DEST"
-# Use the latest Windows Server Core image.
-#
-# WARNING: What's the `:ltsc2019` about?
-# There are two versions of Windows Server 2019:
-# 1. A "regular" one (corresponding to `mcr.microsoft.com/windows/servercore:ltsc2019`)
-# is on a slow release cadence and is the Long-Term Servicing Channel.
-# Mainstream support for this image will end on 1/9/2024.
-# 2. A "fast" release cadence one (corresponding to
-# `mcr.microsoft.com/windows/servercore:1909`) is the Semi-Annual Channel.
-# Mainstream support for this image will end on 5/11/2021.
-#
-# If you choose a different
-# image than described above, change the `:ltsc2019` tag.
-# Start a temporary container in which we install 7-Zip to extract msys2
-FROM mcr.microsoft.com/windows/servercore:ltsc2019 as extract-msys2
-SHELL ["powershell.exe", "-ExecutionPolicy", "Bypass", "-Command", "$ErrorActionPreference='Stop'; $ProgressPreference='SilentlyContinue'; $VerbosePreference = 'Continue';"]
-# Install 7-Zip and add it to the path.
-ADD https://www.7-zip.org/a/7z1801-x64.msi C:\\TEMP\\7z.msi
-RUN Start-Process msiexec.exe -ArgumentList \"/i C:\\TEMP\\7z.msi /qn /norestart /log C:\\TEMP\\7z_install_log.txt\" -wait
-RUN $oldpath = (Get-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name PATH).path; \
-  $newpath = \"$oldpath;C:\Program Files\7-Zip\"; \
-  Set-ItemProperty -Path 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' -Name PATH -Value $newPath
-# Extract msys2
-ADD http://repo.msys2.org/distrib/x86_64/msys2-base-x86_64-20181211.tar.xz C:\\TEMP\\msys2.tar.xz
-RUN 7z x C:\TEMP\msys2.tar.xz -oC:\TEMP\msys2.tar
-RUN 7z x C:\TEMP\msys2.tar -oC:\tools
-# Start building the actual image
+# This Dockerfile creates an image that has the following:
```

Comment:
According to my investigation, just upgrading the docker image won't be enough:
- the `https://github.com/grpc/grpc/tree/master/third_party/toolchains/bazel_0.26.0_rbe_windows` needs to be upgraded as well (as it contains the paths to the VC++ tools etc - so it basically has the VS version hardcoded in it).
- looks like since we're using bazel4.x currently, we'll need to use VS2019 (see comment in the dockerfile).

Reply:
I created https://github.com/grpc/grpc/pull/30072, but still not quite there.